### DISCUSSION TITLE
Introduce the ability to edit chain data.

### DISCUSSION THREAD

**Discussion by neuodev** [wevm/viem#1851] [Idea / Feature Request]
At the moment, you only have two options: 
1. Use existing chains from `viem/chains`.
2. Or define your own chain using `defineChain`.

There are some use cases in which it makes sense to `mutate` the existing chain configs. For example, adding a new default RPC or a new explorer, updating the name of the chain, etc. 

Imagine that the provided RPC in the default chain config doesn't work any more, in this case, the only other option will be to define your full chain config.  

I would like to introduce a new third option: **mutate existing chain data**. 

I can imagine it working like this: 

```ts
import { MuateChain, anvil } from "viem/chains";

const anvil = new MutateChain(anvil)
  .withRpc({ http: "...", socket: "..." })
  .withChainId(-1)
  .withExplorer("...")
  .getChain();
```

Any other implementation will be welcomed. I can implement this feature myself!

**Comment by tmm** [comment]
Chains are just objects. You should be able to manipulate them however you want in user land. Unless this is a common use-case, it not worth designing an API around it yet.
